### PR TITLE
Replaces #1437 - additional documentation for filldist and arraydist

### PR DESCRIPTION
**Pull request by trappmartin**
This PR is aimed to replace #1437.

### CODE REVIEW

**Comment by rikhuijzer** [comment]
Could this PR get a more informative description? `Replaces #1437` doesn't provide much info to the reader of this PR title or to the reader of the Git history once this PR is merged.

File: docs/src/using-turing/guide.md [hunk 1]
```diff
@@ -545,6 +545,41 @@ The `Gibbs` sampler can be used to specify unique automatic differentation backe
...
 For more details of compositional sampling in Turing.jl, please check the corresponding [paper](http://proceedings.mlr.press/v84/ge18b.html).
 
+### Working with filldist and arraydist
+
+Turing provides `filldist(dist::Distribution, n::Int)` and `arraydist(dists::AbstractVector{<:Distribution})` as a simplified interface to construct product distributions, e.g., to model a set of variables that share the same structure but vary by group.
+
+#### Constructing product distributions with filldist
+
+The function `filldist` provides a general interface to construct product distributions over distributions of the same type and parameterisation.
+Note that in contrast to the product distribution interface provided by Distributions.jl (`Product`), `filldist` supports product distributions over univariate or multivariate distributions.
```
Comment (rikhuijzer) [inline docs/src/using-turing/guide.md:555]:
```suggestion
Note that, in contrast to the product distribution interface provided by Distributions.jl (`Product`), `filldist` supports product distributions over univariate or multivariate distributions.
```

File: docs/src/using-turing/guide.md [hunk 2]
```diff
@@ -545,6 +545,41 @@ The `Gibbs` sampler can be used to specify unique automatic differentation backe
...
+
+Example usage:
+
+```julia
+@model function demo(x, g)
+  k = length(unique(g))
+  a ~ filldist(Exponential(), k) # = Product(fill(Exponential(), k))
+  mu = a[g]
+  x .~ Normal.(mu)
```
Comment (rikhuijzer) [inline docs/src/using-turing/guide.md:564]: Wouldn't an `MvNormal` be more performant here?

File: docs/src/using-turing/guide.md [hunk 3]
```diff
@@ -545,6 +545,41 @@ The `Gibbs` sampler can be used to specify unique automatic differentation backe
 
 For more details of compositional sampling in Turing.jl, please check the corresponding [paper](http://proceedings.mlr.press/v84/ge18b.html).
 
+### Working with filldist and arraydist
+
+Turing provides `filldist(dist::Distribution, n::Int)` and `arraydist(dists::AbstractVector{<:Distribution})` as a simplified interface to construct product distributions, e.g., to model a set of variables that share the same structure but vary by group.
```
Comment (devmotion) [inline docs/src/using-turing/guide.md:550]: Just a heads up, rather sooner than later they will be integrated and generalized in Distributions: https://github.com/JuliaStats/Distributions.jl/pull/1391 I am working on a PR to DistributionsAD that would fix the errors mentioned in the last comments; then we can switch `product_distribution` completely from `Product` to `ProductDistribution`.

**Review by yebai — APPROVED** [review]

**Comment by yebai** [comment]
many thanks, @trappmartin!

File: docs/src/using-turing/guide.md (hunk 3 above)
Comment (yebai) [reply docs/src/using-turing/guide.md:550]: Lovely improvements - I look forward to the new Distributions release!

**Pull request merged**